File: app/src/main/java/agersant/polaris/features/collection/BrowseAdapter.java
package agersant.polaris.features.collection;

import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import agersant.polaris.CollectionItem;


abstract class BrowseAdapter
		extends RecyclerView.Adapter<BrowseItemHolder> {

	ArrayList<? extends CollectionItem> items;

	BrowseAdapter() {
		super();
		setItems(new ArrayList<>());
	}

	public void setItems(ArrayList<? extends CollectionItem> items) {
		this.items = items;
		notifyDataSetChanged();
	}

	@Override
	public void onBindViewHolder(BrowseItemHolder holder, int position) {
		holder.bindItem(items.get(position));
	}

	@Override
	public int getItemCount() {
		return items.size();
	}

}

File: app/src/main/java/agersant/polaris/features/collection/BrowseAdapter.kt
package agersant.polaris.features.collection

import agersant.polaris.CollectionItem
import agersant.polaris.ui.CollectionItemDiffUtil
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView

abstract class BrowseAdapter() : RecyclerView.Adapter<BrowseItemHolder>() {

    open var items: List<CollectionItem> = listOf()
        set(value) {
            val diff = DiffUtil.calculateDiff(CollectionItemDiffUtil(field, value))
            field = value
            diff.dispatchUpdatesTo(this)
        }

    override fun onBindViewHolder(holder: BrowseItemHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }
}